src/components/TldrawCanvas.tsx

```diff
@@ -1072,6 +1093,62 @@ const TldrawCanvas = ({ title }: Props) => {
               // @ts-ignore
               static children: typeof TLSelectTool.children = () => {
                 return TLSelectTool.children().map((c) => {
+                  if (c.id === "translating") {
+                    const Translate = c as unknown as typeof Translating;
+                    const allRelationIdSet = new Set(allRelationIds);
+                    const allAddReferencedNodeActionsSet = new Set(
+                      allAddReferencedNodeActions
+                    );
+                    return class extends Translate {
+                      override onPointerUp: TLPointerEvent = () => {
+                        this.onComplete({
+                          type: "misc",
+                          name: "complete",
+                        });
+                        const shape = this.app.getShapeById(this.info.shape.id);
+                        if (!shape) return;
+                        // TODO: find a better way to identify custom arrow shapes
+                        // shape.type or shape.name probably?
```

Comment:
could be useful to hoist this logic up to a helper to consolidate logic we want to improve to one place